src/middleware/validator/user-course.validator.ts

```diff
@@ -0,0 +1,14 @@
+import { check } from 'express-validator'
+
+import validate from './generic.validator'
+
+// TODO: check if user/course id exist
+const userId = check('userId').isNumeric()
+const courseId = check('courseId').isNumeric()
+const level = check('level').isString().trim().isIn(['student', 'ta', 'instructor'])
```

Comment:
This student, ta, instructor bit should probably be a enum somewhere not just here so we use these without having to use the string literal every time. 

It probably belongs in devu-shared tbh. That way we can use it as form options on a user creation form or while displaying shit in the client etc. 

Maybe something like 

```
type UserCourseRole = {
  label: string;
  value: string;
}

const userCourseRoles: UserCourseRole[] = [
  { label: 'TA', value: 'ta' }, 
  ...
]
```

Then here we could use `.IsIn(userCourseRoles.map(u => u.value))`

Reply:
I've updated shared and api with new types. I assume you want the object with label and value for the front end so I created a type in that format like you mention here. I also created a UserCourseLevel type with just the values. Using isIn works/is needed(?) for the validator, but I also wanted to replace this in the UserCourse type and in the model which required a new type.